internal/controllers/iam/userinvitation_controller.go

```diff
@@ -718,3 +729,49 @@ func isUserInvitationExpired(ui *iamv1alpha1.UserInvitation) bool {
 	}
 	return false
 }
+
+func (r *UserInvitationController) updateUserInvitationInviteeUserStatus(ctx context.Context, ui *iamv1alpha1.UserInvitation) error {
+	log := logf.FromContext(ctx).WithName("userinvitation-update-invitee-user-status")
+	log.Info("Updating UserInvitationInviteeUserStatus status", "name", ui.GetName())
+
+	if ui.Status.InviteeUser != nil {
+		log.Info("Invitee User status already set, skipping update", "name", ui.GetName())
+		return nil
+	}
+
+	// Attempt to resolve the invitee user
+	user, err := r.getInviteeUser(ctx, ui.Spec.Email)
+	if err != nil {
+		return fmt.Errorf("failed to get Invitee User: %w", err)
+	}
+
+	// Condition to update the UserInvitation status
+	var cond metav1.Condition
+
+	if user == nil {
+		// Invitee user not found yet
+		cond = metav1.Condition{
+			Type:    inviteeUserStatusUpdateConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  "Invitee User not found",
+			Message: "Invitee User not found, probably not created yet.",
+		}
+	} else {
+		// Invitee user found – update the Status field on the invitation
+		ui.Status.InviteeUser = &iamv1alpha1.UserInvitationInviteeUserStatus{
+			Name: user.Name,
+		}
+		cond = metav1.Condition{
+			Type:    inviteeUserStatusUpdateConditionType,
+			Status:  metav1.ConditionTrue,
+			Reason:  "Updated",
+			Message: "Invitee User found",
+		}
```

Comment:
Can we make the text here a little more descriptive / human friendly? 

```suggestion
		cond = metav1.Condition{
			Type:    inviteeUserStatusUpdateConditionType,
			Status:  metav1.ConditionTrue,
			Reason:  "InvitedUserRegistered",
			Message: "Confirmed the invited user has an account registered with the platform.",
		}
```